Jackal/Jackal Shared/Playing Field/Shared/Level.swift

```diff
@@ -48,33 +48,25 @@ class Level {
         
         for x in 0..<Int8(size.width) {
             for y in 0..<Int8(size.height) {
-                let position = BoardPosition(x, y)
-                let node = BoardGraphNode(boardPosition: position)
-                graph.add([node])
+                let fieldNode = fieldNodes[Int(x)][Int(y)]
+                fieldNode.nodeConnector.createNodes(fieldNode: fieldNode, graph: graph, x: x, y: y)
             }
         }
         
-        for x in 0..<Int8(configuration.size.width) {
+        for x in 0..<Int8(size.width) {
             for y in 0..<Int8(size.height) {
-                
                 let fieldNode = fieldNodes[Int(x)][Int(y)]
-                let position = BoardPosition(x, y)
-                guard let centre = graph.node(at: position) else { continue }
-                
-                let connectedPositions: [BoardPosition]
-                
-                switch fieldNode.moveType {
-                case .any:
-                    connectedPositions = []//TODO: Fix
-                    
-                case .oneOf(let moves):
-                    connectedPositions = moves.map { BoardPosition(x + $0.x, y + $0.y) }
-                }
-                
-                let connectedNodes = connectedPositions.flatMap(graph.node(at:))
-                centre.addConnections(to: connectedNodes, bidirectional: false)
+                fieldNode.nodeConnector.addNodesConnections(fieldNode: fieldNode, graph: graph, x: x, y: y)
             }
         }
+        
+        for x in 0..<Int8(size.width) {
+            for y in 0..<Int8(size.height) {
```

Comment:
So, now I have question, why do we use 2 loops?

Reply:
Actually we have 3 iterations:
1) Create nodes
2) Connect all nodes
3) Remove unwanted connections

Reply:
3 loops

Reply:
First adds nodes
Second adds connections
Third removes connections